image_buffer/image_buffer.go

```diff
@@ -130,13 +149,88 @@ func (ib *ImageBuffer) PopFirstToSend() (CachedData, bool) {
 	ib.mu.Lock()
 	defer ib.mu.Unlock()
 	if len(ib.toSend) == 0 {
+		if ib.debug {
+			ib.logger.Infow("PopFirstToSend buffer empty",
+				"method", "PopFirstToSend",
+				"toSendSize", 0)
+		}
 		return CachedData{}, false
 	}
 	x := ib.toSend[0]
 	ib.toSend = ib.toSend[1:]
+
+	// Apply timestamp naming to the images
+	x.Imgs = timestampImagesToNames(x.Imgs, x.Meta)
+
+	if ib.debug {
+		remainingLen := len(ib.toSend)
+		ib.logger.Infow("PopFirstToSend consumed image",
+			"method", "PopFirstToSend",
+			"imagesConsumed", 1,
+			"remainingToSendSize", remainingLen)
+	}
 	return x, true
 }
 
+// timestampImagesToNames converts images to have timestamp-based names in format "[timestamp]_[original_name]"
+func timestampImagesToNames(images []camera.NamedImage, meta resource.ResponseMetadata) []camera.NamedImage {
+	result := make([]camera.NamedImage, len(images))
+	for i, img := range images {
+		result[i] = img // Copy the image
+
+		// Use timestamp as prefix - use "no-date" if timestamp not available
+		timestampStr := noDateString
+		if !meta.CapturedAt.IsZero() {
```

Comment:
Can you explain why need this check?

Reply:
This is just to check if the time for some reason time was set to t=0. That defaults to midnight Jan 1, 1970 UTC. 
When checking for invalid times, you can either parse the time and make sure to filter out the Jan 1 1970 entries, but making it just a string called "no-date" would trip up any parsing function much more obviousy